ambry-api/src/main/java/com/github/ambry/store/MessageErrorInfo.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2024 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+public class MessageErrorInfo {
```

Comment:
@DevenWalia We will need this class to be public since it is being accessed outside the com.github.ambry.store package.
Agreed on the lombok annotations, will look to including this new dependency and integrate it atleast for the new classes.

Reply:
Yea, wherever possible, try to use package-private access for new classes.
There might be a whole lot of places where this cannot be done due to the earlier way of packaging classes, which is fine.